### PR TITLE
feat: Support environment variables in Helm value file paths

### PR DESCRIPTION
**Pull request by mariussturm**
Hey Team!
Just a tiny PR that adds support for env vars in the Helm value file paths.

I have a use case where I need access to ARGOCD_APP_NAME and ARGOCD_APP_REVISION in combination with a custom protocol. The Helm configuration would look like this:

```
helm:
    valueFiles:
      - values.yaml
      - myprotocol://somepath/$ARGOCD_APP_NAME/$ARGOCD_APP_REVISION
```

Env vars are already supported for Helm `parameters` but not for `valueFiles`, this PR adds this functionality.
 
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by crenshaw-dev** [comment]
@mariussturm this looks good! Can you resolve the DCO check?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Would you mind adding a small note to the docs about this feature? https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#values-files

File: docs/user-guide/helm.md [hunk 1]
```diff
@@ -164,6 +164,17 @@ Or via declarative syntax:
...
 
+It's also possible to use build environment variables for the Helm values file path:
+
+```yaml
+  spec:
+    source:
+      helm:
+        valueFiles:
+              - values.yaml
+              - myprotocol://somepath/$ARGOCD_APP_NAME/$ARGOCD_APP_REVISION
```
Comment (crenshaw-dev) [inline docs/user-guide/helm.md:175]: Tiiiiny nitpick. Could the indentation be shifted left? I think this is valid yaml, just looks a bit odd.  :-)

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: docs/user-guide/helm.md (hunk 1 above)
Comment (mariussturm) [reply docs/user-guide/helm.md]: Good point! aligned it now with the parameters example above.

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks @mariussturm!

**Comment by mariussturm** [comment]
Awesome, thanks for the super fast review @crenshaw-dev 💪 💯

**Pull request merged**